tests/model_inheritance/models.py

```diff
@@ -186,3 +186,7 @@ class Child(Parent):
 
 class GrandChild(Child):
     pass
+
+
+class ItalianRestaurantManyParents(ItalianRestaurant, Place):
+    place_ptr_2 = models.OneToOneField(Place, models.CASCADE, parent_link=True)
```

Comment:
This is not a proper model for a regression test as it doesn't crash with `AttributeError: 'OneToOneField' object has no attribute 'attname'`.

Reply:
Please check https://github.com/django/django/pull/16952#issuecomment-1581806139

Reply:
OK, so this PR contains some experiments around MTI it's not a fix for 34634.

Reply:
@felixxm The bug seems much bigger than onetoonefield not having attname. The same test in the PR passing using postgres in my local, let me know your configuration if you can reproduce it locally. Because this behavior seems totally weird to me, I am nowhere close to any explanation for this.

Reply:
@Akash-Kumar-Sen Please don't constantly ping folks. They are already aware that you're working on this ticket. You're picking mostly complicated ORM issues so you cannot expect people to come up with ready to go solutions. For example, I don't have an immediate answer for you.

Reply:
Sorry for that, actually I thought GitHub UI might be tricky enough for people to miss comments, like last time you overlooked my comment about experiments.